pkg/trace/stats/sublayers.go

```diff
@@ -34,6 +33,90 @@ func (v SublayerValue) GoString() string {
 	return v.String()
 }
 
+type SublayerCalculator struct {
```

Comment:
Can you please document all of this thoroughly? It's an open-source project and many people will look at this code trying to understand how it works.

Reply:
Please explain how each field is used in the algorithm.